src/swdmt/redistricting/RegionTest.java

```diff
@@ -45,6 +47,24 @@ public void setUp() {
     public void tearDown() {
     }
 
+    /**
+     * Test case for Region(Collection<Voter>)
+     * Should return True
+     */
+    @Test
+    public void voterRegionInputConstructorTest() {
+        Voter voterA = new Voter(Party.PARTY0, new Location(0, 0));
+        Voter voterB = new Voter(Party.PARTY1, new Location(1, 1));
+        Collection<Voter> voters = new HashSet<Voter>();
+        voters.add(voterA);
+        voters.add(voterB);
+        Region region = new Region(voters);
+        for (Voter v : voters) {
+            assertTrue(region.voters().contains(v));
+            assertTrue(region.locations().contains(v.location()));
```

Comment:
Also need to verify that no undesired voters or locations appear in the region.

Reply:
Given the simple single case this test addresses the shortcoming is minor and the test will be accepted even though incomplete.